ocs_ci/utility/vsphere.py

```diff
@@ -1333,3 +1312,56 @@ def find_vms_without_ip_and_restart(self):
             logger.info(f"VM: {vm.name} doesn't have IP, it will be restarted!")
         if vms_without_ip:
             self.restart_vms(vms_without_ip, force=True)
+
+    def get_device_by_key(self, vm, identifier, key="unit_number"):
```

Comment:
good idea to move this into a new method

Reply:
yes